src/angular/app.js

```diff
@@ -228,9 +249,64 @@ controllers.controller('ResultsCtrl', ['$scope', 'Search', function ($scope, Sea
 
         return activityDetails;
     }
+
+    $scope.setServiceId = function (serviceId) {
+    	// var url = '/#/services/' + serviceId;
+    	$location.path('/services/' + serviceId);
+    }
 }]);
 
 controllers.controller('NavbarCtrl', ['$scope', function ($scope) {}]);
 
+controllers.controller('ServiceCtrl', ['$scope', '$routeParams', 'Search', function($scope, $routeParams, Search) {
+	var servicesById = Search.getServicesById();
```

Comment:
Nice work, so instead of using `Search.getServicesById` use `ServiceRepository.findById($routeParams.serviceId)`